realm/realm-library/src/objectServer/java/io/realm/SyncConfiguration.java

```diff
@@ -464,8 +463,14 @@ public Builder encryptionKey(byte[] key) {
         }
 
         /**
-         * Sets the schema version of the Realm. This must be equal to or higher than the schema version of the existing
-         * Realm file, if any. If the schema version is higher than the already existing Realm, a migration is needed.
+         * Sets the schema version of the Realm.
+         * <p>
+         * While synced Realms only support additive schema changes which can be applied without requiring a manual
+         * migration, the schema version must still be incremented as an indication to Realm that the change was
+         * intentional.
+         * <p>
+         * Failing to increment the schema version will cause Realm to throw a {@link io.realm.exceptions.RealmMigrationNeededException}
+         * when the Realm is opened and the changed schema will not be applied.
```

Comment:
What if the server side Realm schema version is higher than the client side? That needs to be documented here as well.